packages/discovery-provider/src/queries/download_csv.py

```diff
@@ -72,12 +73,19 @@ def get_purchases_or_sales(user_id: int, is_purchases: bool):
                     USDCPurchase.country.label("country"),
                     User.name.label("buyer_name"),
                     EncryptedEmail.encrypted_email.label("encrypted_email"),
+                    EmailAccess.encrypted_key.label("encrypted_key"),
```

Comment:
this query is confuses me a bit.

in my head it would just be

```
select *
from email_access
join usdc_purchase on usdc_purchase.buyer_user_id = email_access.email_owner_user_id
where email_access.seller_user_id = user_id
```
what am i missing?

Reply:
we have two cases 

- The seller (user_id) is either the receiving_user_id (direct access)
- OR where they are the grantor_user_id (they granted access to someone else)

so the query should look a little like this, updated to reflect 

```
SELECT *
FROM email_access
JOIN usdc_purchase ON usdc_purchase.buyer_user_id = email_access.email_owner_user_id
WHERE email_access.receiving_user_id = :user_id 
   OR email_access.grantor_user_id = :user_id;
```

Reply:
yeah ok that makes sense. maybe just not reading the sqlalchemy code well (i hate this orm nonsense lol).

feels like we shouldn't need an outer join though